ui/index.d.ts

```diff
@@ -1182,12 +1190,21 @@ export interface UIExtensionVariables {
     outputChannel: IAzExtOutputChannel;
     ui: IAzureUserInput;
 
+    experimentationService?: IExperimentationServiceAdapter;
+
     /**
      * Set to true if not running under a webpacked 'dist' folder as defined in 'vscode-azureextensiondev'
      */
     ignoreBundle?: boolean;
 }
 
+export interface IExperimentationServiceAdapter {
+    isCachedFlightEnabled(flight: string): Promise<boolean>;
+    isLiveFlightEnabled(flight: string): Promise<boolean>;
+    getCachedTreatmentVariable<T extends string | number | boolean>(name: string): T | undefined;
```

Comment:
Can you give some quick examples on how these are used, maybe an existing Docker scenario? If Docker isn't using all these methods, I would err on the side of not supporting them until someone actually wants them.

Reply:
Cached vs live: Cached will grab the value from the cache, which means it's ~1 session delayed. TAS will "fill" its in-memory cache, then load from the web and update the disk cache, but _not_ the in-memory cache (otherwise, users could see behaviors change mid-session). So, the next session will get the new data. Live, however, goes straight to the web and forces an in-memory cache update, and also takes longer. This is useful if you need to avoid the 1 session delay.

Flights: simple yes/no of whether the flight is enabled.

Treatment variables: Any string, number, or boolean can be passed (you could also implement boolean flighting with this if you wanted to). For example, you may want to have two treatment groups and a control group, each with a different string shown somewhere in the UI, to compare the results.

I'll add the above to comments in the code. We do have plans to use the treatment variables in the near-ish future so I figured it would be best to just implement it now.